app/main.py

```diff
@@ -1,3 +1,109 @@
-def shop_trip():
-    # write your code here
-    pass
+import json
+import math
+
+from app.car import Car
+from app.customer import Customer
+from app.shop import Shop
+
+
+def get_dict_from_json() -> dict:
+    with open("app/config.json", "r") as data:
+        data = json.load(data)
+
+        customers = []
+        for customer in data["customers"]:
+            customers.append(Customer(
+                name=customer["name"],
+                product_cart=customer["product_cart"],
+                location=customer["location"],
+                money=customer["money"],
+                car=Car(brand=customer["car"]["brand"],
+                        fuel_consumption=customer["car"]["fuel_consumption"], )
+            ))
+
+        shops = []
+        for shop in data["shops"]:
+            shops.append(Shop(
+                name=shop["name"],
+                products=shop["products"],
+                location=shop["location"],
+            ))
+        return {
+            "FUEL_PRICE": data["FUEL_PRICE"],
+            "customers": customers,
+            "shops": shops,
+        }
```

Comment:
u don't need to implement all logic in a read stream
shift indent after reading data one level to the left